.github/workflows/build.yml

```diff
@@ -2,33 +2,57 @@
 name: build
 
 on:
-  # TODO build also on PRs
+  pull_request:
   push:
     branches: [main]
-
+    tags:
+      - '*'
 # This job installs dependencies, build the book, and pushes it to `gh-pages`
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.6.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: Checkout
+      uses: actions/checkout@v2
 
     # Install dependencies
     - name: Set up Python 3.8
       uses: actions/setup-python@v1
       with:
         python-version: 3.8
+
     - name: Install dependencies
       run: |
         pip install msprime --pre # Needed until we have a stable 1.0
         pip install -r requirements.txt
+
     # Build the book
     - name: Build the book
       run: ./build.sh
 
     # Push the book's HTML to github-pages
     - name: GitHub Pages action
+      if: github.ref == 'refs/heads/main'
       uses: peaceiris/actions-gh-pages@v3.5.9
       with:
         github_token: ${{ secrets.GITHUB_TOKEN }}
         publish_dir: ./_build/html
+
+    - name: Sleep for 3 min to allow github pages to process the site
```

Comment:
I had thought the push to the gh-pages branch was immediate, it was the followup stuff where the contents of this branch got processed by jekyll and served on the actual front-end was where the delay is. We're pulling from the gh-pages branch, rather than the final http destintation, aren't we?

Reply:
Yep, you're right! Removed.

Reply:
sleep() calls make me twitch - hangover from my async server programming days!